pallas-txbuilder/src/transaction/model.rs

```diff
@@ -0,0 +1,712 @@
+use pallas_addresses::Address as PallasAddress;
+use pallas_crypto::{
+    hash::{Hash, Hasher},
+    key::ed25519,
+};
+use pallas_primitives::{babbage, Fragment};
+
+use std::{collections::HashMap, ops::Deref};
+
+use serde::{Deserialize, Serialize};
+
+use crate::TxBuilderError;
+
+use super::{
+    AssetName, Bytes, Bytes32, Bytes64, DatumBytes, DatumHash, Hash28, PolicyId, PubKeyHash,
+    PublicKey, ScriptBytes, ScriptHash, Signature, TransactionStatus, TxHash,
+};
+
+// TODO: Don't make wrapper types public
+#[derive(Default, Serialize, Deserialize, PartialEq, Eq, Debug)]
+pub struct StagingTransaction {
+    pub version: String,
+    pub status: TransactionStatus,
+    pub inputs: Option<Vec<Input>>,
+    pub reference_inputs: Option<Vec<Input>>,
+    pub outputs: Option<Vec<Output>>,
+    pub fee: Option<u64>,
+    pub mint: Option<MintAssets>,
+    pub valid_from_slot: Option<u64>,
+    pub invalid_from_slot: Option<u64>,
+    pub network_id: Option<u8>,
+    pub collateral_inputs: Option<Vec<Input>>,
+    pub collateral_output: Option<Output>,
+    pub disclosed_signers: Option<Vec<PubKeyHash>>,
+    pub scripts: Option<HashMap<ScriptHash, Script>>,
+    pub datums: Option<HashMap<DatumHash, DatumBytes>>,
+    pub redeemers: Option<Redeemers>,
+    pub script_data_hash: Option<Bytes32>,
+    pub signature_amount_override: Option<u8>,
+    pub change_address: Option<Address>,
+    // pub certificates: TODO
+    // pub withdrawals: TODO
+    // pub updates: TODO
+    // pub auxiliary_data: TODO
+    // pub phase_2_valid: TODO
```

Comment:
Oh this is exciting, let's start moving Aiken's uplc crate to pallas-plutus when we start building this part? 

cc @KtorZ